c/src/core/data.h

```diff
@@ -62,6 +62,8 @@ struct pn_data_t {
   pni_nid_t base_current;
 };
 
+pn_data_t *pni_data(size_t capacity, size_t initial_buffer);
```

Comment:
As above - unless there is a measurable performance penalty I don't think this is necessary.